btree/btree.c

```diff
@@ -1977,15 +2014,30 @@ static int64_t mem_node_get(struct node_op *op, struct td *tree,
 	return nxt_state;
 }
 
-static void mem_node_put(struct nd *node)
+static void mem_node_put(struct nd *node, struct td *tree)
 {
-	/**
-	 * This implementation does not perform any action, but the final one
-	 * should decrement the reference count of the node and if this
-	 * reference count goes to '0' then detach this node descriptor from the
-	 * existing node and make this node descriptor available for 'node_get'.
-	 */
 	node->n_ref--;
```

Comment:
don't we want to have implementation of nd::n_lock in this PR, because you are dealing with adding and deleting nd from ndlist /tree list , there can be scenario where one thread see node->n_ref ==0 and meanwhile other thread increase n_ref, So, to avoid such scenarios we should  increment or decrement n_ref using n_lock and should perform addition /deletion of nd from ndlist/treelist

Reply:
I wanted to but since the n_lock will be used for other fields of nd also (n_seq) etc, I thought of taking it as a separate task. I will do one thing, will add n_ref in this PR.  Rest of the n_lock related changes can be taken up in separate task.